docs/internal-documentation/maintenance/SystemTesting.md

```diff
@@ -0,0 +1,211 @@
+System testing
+==============
+Before a new version of Zonemaster can be released it must pass three levels of
+system testing: Installation Testing, Smoke Testing and Acceptance Testing.
+Each test level is described in a separate section below. A separate section is
+reserved for the concept of Configurations, which is used by other sections.
+
+
+1. Configurations
+-----------------
+The purpose of configurations is to provide different contexts for the tests to
+run in.
+
+A configuration is a tuple of the following parameters:
+
+* architecture (e.g. x86_64)
+* operating system version (e.g. Ubuntu 14.04)
+* database engine version (e.g. PostgreSQL 9.3)
+* system locale (e.g. en_US.UTF-8)
+
+Each parameter has a section below definig the parameters more closely.
+
+The set of possible values for each configuration parameter is declared in the
+README.md of the top-level zonemaster repository.
+
+Each test level section specifies a subset of all possible configuration tuple
+to test.
+
+
+### Architecture
+
+The only architecture we support is amd64/x86_64.
+
+
+### Operating system version
+
+The following operating systems are supported:
+
+* Centos
+* Debian
+* FreeBSD
+* Ubuntu
+
+For each supported operating system, the supported versions are determined using
+the following criteria.
+
+* CentOS:
+  * The supported versions are the Base Distributions (excluding 5 and 6) as
+    specified at: 
+    https://en.wikipedia.org/wiki/CentOS#End-of-support_schedule
+  * The supported minor release is the one specified at:
+    https://wiki.centos.org/Download
+
+* Debian:
+  * The supported versions are "stable" and "oldstable" as specified at: 
+    https://wiki.debian.org/DebianReleases#Current_Releases.2FRepositories
+  * The supported point release is the current one as specified at:
+    https://wiki.debian.org/DebianReleases/PointReleases
+
+* FreeBSD:
+  * The supported versions are the Extended releases (excluding 9.3) as
+    specified at: https://www.freebsd.org/security/
+  * Patch level should not be specified.
+
+* Ubuntu:
+  * The supported versions are the LTS releases (excluding 12.04) as specified
+    at: https://wiki.ubuntu.com/Releases
+  * The supported patch level is the one specified at:
+    https://wiki.ubuntu.com/Releases
+
+
+### Database engine version
+
+The following database engines are supported:
+
+* MySQL
+* PostgreSQL
+* SQLite
+
+... To be specified. It's slightly complicated.
+
+
+### System locale
+
+Zonemaster promises to behave correctly when the system locale is set to one of
+these locales.
+
+... Do we support all possible locales in this sense? Since we cannot reasonably
+perform testing for all possible locales, supported system locales are grouped
+into a small set of "tier 1" system locales to be used in test configurations,
+and the remaining "tier 2" system locales.
+
```

Comment:
I think it is reasonably to support only en_US.utf8 and C for installation and running the backend and GUI. That keeps those parts down. And to support at least en_US.utf8, C, fr_FR.utf8 and sv_SE.utf8 for Zonemaster CLI. I would like to see it expanded to en_XX.utf8, C, C.utf8, fr_XX.utf8 and sv_XX.utf8. Changing locale when running Zonemaster CLI commands is managable.